src/System.Private.CoreLib/shared/System/Text/Unicode/Utf8Utility.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Unicode
+{
+    internal static class Utf8Utility
+    {
+        /// <summary>
+        /// The UTF-8 representation of <see cref="UnicodeUtility.ReplacementChar"/>.
+        /// </summary>
+        private static ReadOnlySpan<byte> ReplacementCharSequence => new byte[] { 0xEF, 0xBF, 0xBD };
+
+        /// <summary>
+        /// Returns the byte index in <paramref name="utf8Data"/> where the first invalid UTF-8 sequence begins,
+        /// or -1 if the buffer contains no invalid sequences. Also outs the <paramref name="isAscii"/> parameter
+        /// stating whether all data observed (up to the first invalid sequence or the end of the buffer, whichever
+        /// comes first) is ASCII.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int GetIndexOfFirstInvalidUtf8Sequence(ReadOnlySpan<byte> utf8Data, out bool isAscii)
+        {
+            // TODO: Make this slow implementation faster.
```

Comment:
This kind of comment/TODO is less useful and/or noise since it is a general - "make things faster". The previous few mentioned a specific approach to get better perf, is there one here?

Reply:
https://github.com/dotnet/coreclr/pull/21948 is an optimized drop-in replacement for this method. I should update the comment to call this out explicitly (and use `TODO_UTF8STRING`, as you suggested elsewhere).